ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/versions/deneb/util/AttestationUtilDenebTest.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright ConsenSys Software Inc., 2022
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.spec.logic.versions.deneb.util;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static tech.pegasys.teku.infrastructure.time.TimeUtilities.secondsToMillis;
+
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.SpecMilestone;
+import tech.pegasys.teku.spec.SpecVersion;
+import tech.pegasys.teku.spec.TestSpecFactory;
+
+public class AttestationUtilDenebTest {
+
+  private static final SpecVersion SPEC_VERSION =
+      TestSpecFactory.createMinimalDeneb().forMilestone(SpecMilestone.DENEB);
+
+  private static final int SECONDS_PER_SLOT = SPEC_VERSION.getConfig().getSecondsPerSlot();
+
+  private final AttestationUtilDeneb attestationUtilDeneb =
+      (AttestationUtilDeneb) SPEC_VERSION.getAttestationUtil();
+
+  @ParameterizedTest
+  @MethodSource("provideAttestationSlotInCurrentOrPreviousEpochArguments")
+  public void testAttestationSlotInCurrentOrPreviousEpoch(
+      final int attestationSlot, final UInt64 currentTime, final boolean expectedResult) {
+    final UInt64 currentTimeMillis = secondsToMillis(currentTime);
+    // set genesisTime as 0 for simplification
+    final UInt64 genesisTime = UInt64.ZERO;
+    final boolean actualResult =
+        attestationUtilDeneb.isAttestationSlotInCurrentOrPreviousEpoch(
+            UInt64.valueOf(attestationSlot), genesisTime, currentTimeMillis);
+    assertThat(actualResult).isEqualTo(expectedResult);
+  }
+
+  // minimal is 6 seconds per slot, 8 slots per epoch
+  private static Stream<Arguments> provideAttestationSlotInCurrentOrPreviousEpochArguments() {
+    return Stream.of(
+        Arguments.of(6, getTimeForSlot(10), true),
+        Arguments.of(6, getTimeForSlot(3), true),
+        Arguments.of(6, getTimeForSlot(26), false),
```

Comment:
Boundary type tests can be more readable testing the slot we fail on vs the slot we pass on...
so for example, at 8 slots per epoch, (so 0-7, 8-15, 16-23), test slot 15,16 etc... (one in second epoch, one the slot after) - helps see what we're testing sometimes, or pass a name where you could describe is sometimes done, so then you could go `upper_bound`, `lower_bound` kind of idea...

Reply:
makes sense, I added comments and improved test cases across the two test classes. Think it's more clear now and covers more cases.